TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/http/retries/DefaultRetry.java

```diff
@@ -0,0 +1,54 @@
+package org.opensearch.migrations.replay.http.retries;
+
+import java.util.List;
+
+import org.opensearch.migrations.replay.AggregatedRawResponse;
+import org.opensearch.migrations.replay.IRequestResponsePacketPair;
+import org.opensearch.migrations.replay.RequestSenderOrchestrator;
+import org.opensearch.migrations.replay.util.TextTrackedFuture;
+import org.opensearch.migrations.replay.util.TrackedFuture;
+
+import io.netty.buffer.ByteBuf;
+
+public class DefaultRetry implements RequestRetryEvaluator {
+    static final int MAX_RETRIES = 4;
+
+    public static boolean retryIsUnnecessaryGivenStatusCode(int statusCode) {
+        switch (statusCode) {
+            case 200:
+            case 201:
+            case 401:
+            case 403:
+                return true;
+            default:
+                return statusCode >= 300 && statusCode < 400;
+        }
+    }
+
+    public TextTrackedFuture<RequestSenderOrchestrator.RetryDirective>
+    getRetryDirectiveUnlessExceededMaxRetries(List<AggregatedRawResponse> previousResponses) {
+        var d = previousResponses.size() > MAX_RETRIES ?
```

Comment:
nit: is current response in previous retries? If not, this should be a `>=`

 (3 previousResponses + current response == 4 calls == 1 original call and 3 retries)